src/openrct2/ride/Ride.cpp

```diff
@@ -3451,45 +3451,76 @@ static TrainReference VehicleCreateTrain(
     return train;
 }
 
-static bool VehicleCreateTrains(Ride& ride, const CoordsXYZ& trainsPos, TrackElement* trackElement)
+static bool VehicleCreateTrains(Ride& ride)
 {
     TrainReference firstTrain = {};
     TrainReference lastTrain = {};
-    int32_t remainingDistance = 0;
     bool allTrainsCreated = true;
 
-    for (int32_t vehicleIndex = 0; vehicleIndex < ride.NumTrains; vehicleIndex++)
+    std::vector<StationIndex> validStationIds;
+    for (uint8_t i = 0; i < static_cast<uint8_t>(std::size(ride.GetStations())); i++)
```

Comment:
I'm not sure the cast is needed it should be only a problem with signedness is different.
```suggestion
    const auto stationCount = ride.GetStations().size();
    for (uint8_t i = 0; i < stationCount ; i++)
```

Reply:
I ran into problems with MSVC rejecting my commits. Is this something you could fix locally and push to my branch? Otherwise it will involve pushing a lot of commits blindly to the CI in the hope MSVC will accept it :')